### ISSUE TITLE
semantic-release-backmerge not working from v1.5.1 with Gitlab-ci.

### ISSUE THREAD

**Issue by bychis** [saitho/semantic-release-backmerge#21]
Hello.
semantic-release-backmerge not working from v1.5.1 with Gitlab-ci.

* My config is this (.releaserc.json)
```
{
    "branches": ["main"],
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/gitlab",
      "@semantic-release/npm",
      "@semantic-release/changelog",
      [
        "@semantic-release/git",
        {
          "assets": ["package.json","CHANGELOG.md","yarn.lock"],
          "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
        }
      ],
      [
        "@saithodev/semantic-release-backmerge",
        {
          "branchName": "dev",
          "clearWorkspace": "true",
          "plugins": [
            [
              "@semantic-release/exec",
              {
                "successCmd": "echo 'Version in master is ${nextRelease.version}' > test.txt && git add test.txt"
              }
            ]
          ]
        }
      ]
    ]
  }
```

* my .gitlab-ci.yml config is this
```
before_script:
  - sudo chown $USER:$USER . 
  - chmod +x ./env.sh
  - ./env.sh
stages:
  - build
  - test
  - release
  - publish
  - deploy

deploy on staging:
  stage: build 
  script:
    - docker image prune -f
    - docker-compose down
    - docker-compose up -d --build
  only:
    - dev
  tags:
    - dev-server

publish release:
  stage: release
  variables:
    GITLAB_TOKEN: $GITLAB_TOKEN
  script:
    - npx semantic-release
  only:
    - main
  tags:
    - dev-server

build a production-ready docker image:
  stage: publish
  script:
    - echo "Build a production-ready docker image ..."
    - docker build -f Dockerfile.prod -t ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA} .
  needs: ['publish release']
  only:
    - main
  tags:
    - dev-server

deploy the new docker image:
  stage: deploy
  script:
    - docker rm -f kio-showcase
    - docker run -d -p --name kio-showcase 127.0.0.1:3000:3000 ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}
  needs: ['build a production-ready docker image']
  only:
    - main
  tags:
    - prod-server
  when: manual

```

* GItlab-ci errors:
```
[13:04:39] [semantic-release] [@saithodev/semantic-release-backmerge] › ℹ  Release succeeded. Performing back-merge into branch "dev".
[13:04:39] [semantic-release] [@saithodev/semantic-release-backmerge] › ℹ  Stashing uncommitted files from Git workspace.
[13:04:39] [semantic-release] › ✖  Failed step "success" of plugin "@saithodev/semantic-release-backmerge"
[13:04:39] [semantic-release] › ✖  An error occurred while running semantic-release: { Error: Command failed with exit code 1: git checkout dev
error: pathspec 'dev' did not match any file(s) known to git
    at makeError (/usr/local/lib/node_modules/@saithodev/semantic-release-backmerge/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/usr/local/lib/node_modules/@saithodev/semantic-release-backmerge/node_modules/execa/index.js:118:26)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  shortMessage: 'Command failed with exit code 1: git checkout dev',
  command: 'git checkout dev',
  escapedCommand: 'git checkout dev',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr:
   'error: pathspec \'dev\' did not match any file(s) known to git',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false,
  pluginName: '@saithodev/semantic-release-backmerge' }
{ AggregateError: 
    Error: Command failed with exit code 1: git checkout dev
    error: pathspec 'dev' did not match any file(s) known to git
        at makeError (/usr/local/lib/node_modules/@saithodev/semantic-release-backmerge/node_modules/execa/lib/error.js:60:11)
        at handlePromise (/usr/local/lib/node_modules/@saithodev/semantic-release-backmerge/node_modules/execa/index.js:118:26)
    at /usr/local/lib/node_modules/semantic-release/lib/plugins/pipeline.js:54:11
    at process._tickCallback (internal/process/next_tick.js:68:7) name: 'AggregateError' }
Cleaning up file based variables
00:01
ERROR: Job failed: exit status 1
```

**Comment by saitho** [comment]
@bychis Can I assume that v1.5.0 worked fine?
I'll take a look at it.

**Comment by bychis** [comment]
@saitho Yes it did

**Comment by saitho** [comment]
@bychis The local branches were gone after checking out the repository via the URL.
Should be solved in v1.5.2. :)
